### ISSUE TITLE
issue running cmake with clang installed on lib64 folder instead of lib

### ISSUE THREAD

**Issue by chunzhu** [CoatiSoftware/Sourcetrail#1058]
Hi,

I am using fedora 32.
llvm and clang are both installed with dnf.

When i run cmake instructions as per [readme](https://github.com/CoatiSoftware/Sourcetrail/blob/master/README.md#how-to-build).
It issued error Unable to copy Clang compiler headers from clang build dir.

Digging into the issue, i realised that CLANG_COMPILER_HEADER_SEARCH_PATH was looking into /lib folder instead of /lib64. it seems that Clang compiler headers are installed onto lib64 folder.

i changed the static path "${LLVM_BINARY_DIR}/${LLVM_CONFIGURATION_TYPE}/lib/clang/${LLVM_VERSION}/include" to  "${LLVM_BINARY_DIR}/${LLVM_CONFIGURATION_TYPE}/lib64/clang/${LLVM_VERSION}/include" and everything seems working, the cmake  and make were successful and i can run the sample files .

is there a better approach so cmake can determine which folder lib or lib64 to set CLANG_COMPILER_HEADER_SEARCH_PATH?

**Comment by mlangkabel** [comment]
Hey @chunzhu, thanks for reporting! Can you please try to print the value of the `LLVM_BUILD_LIBRARY_DIR` from within Sourcetrail's CMakeLists.txt file (after the `find_package(Clang REQUIRED)` line and tell us the value? Maybe I can change the CMake to be based on that variable.

**Comment by snikulov** [comment]
Same issue on Fedora 32 (64bit)
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index b60eaaa6..b5bc2684 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,7 +126,7 @@ endif()
 if (BUILD_CXX_LANGUAGE_PACKAGE)
 
        find_package(Clang REQUIRED)
-
+      message(STATUS "LLVM_BUILD_LIBRARY_DIR = ${LLVM_BUILD_LIBRARY_DIR}")
        if (LLVM_FOUND)
                message(STATUS "Found LLVM ${LLVM_VERSION}")
        endif()
```
Output is
```
-- Version: 2020.2.48
-- setting up the buildtype for versioning
-- ccache found
-- 'Treat warnings as errors' policy is disabled
-- LLVM_BUILD_LIBRARY_DIR = 
-- Found LLVM 10.0.0
-- Trying to find Clang compiler headers in '.' build config in directory '/usr/./lib/clang/10.0.0/include'.
CMake Error at CMakeLists.txt:156 (message):
  Unable to copy Clang compiler headers from clang build dir.


-- Configuring incomplete, errors occurred!
See also "/home/snikulov/work/github/Sourcetrail/build/CMakeFiles/CMakeOutput.log".
```

While headers is located in 
`/usr/lib64/clang/10.0.0/include/`

**Comment by anktkr** [comment]
@mlangkabel Wouldn't it be more robust to directly ask users for the header search path, instead of deriving it from llvm build directories.

For me, on macOS, this goes very wrong. 
```
-- Found LLVM 11.0.0git
Trying to find Clang compiler headers in '.' build config in directory '/Users/me/Applications/llvm-project-master/binstall/./lib/clang/11.0.0git/include'.
CMake Error at CMakeLists.txt:156 (message):
  Unable to copy Clang compiler headers from clang build dir.
```
with the cmakecache variables:
`LLVM_DIR:PATH=/Users/me/Applications/llvm-project-master/binstall/lib/cmake/llvm`
`Clang_DIR:PATH=/Users/me/Applications/llvm-project-master/binstall/lib/cmake/clang`

The only two paths under the root directory which contain version number in them are 
`/Users/me/Applications/llvm-project-master/binstall/lib/clang/11.0.0/include` this has the headers.
`/Users/me/Applications/llvm-project-master/build/lib/clang/11.0.0/include` this only has some folders with a few headers.

I built by changing the cmakelists.txt to this 
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index b60eaaa6..2dcbd9f0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,11 +135,7 @@ if (BUILD_CXX_LANGUAGE_PACKAGE)
 		set(LLVM_CONFIGURATION_TYPES ".")
 	endif()
 
-	set (CLANG_COMPILER_HEADER_SEARCH_PATH "")
-	foreach(LLVM_CONFIGURATION_TYPE ${LLVM_CONFIGURATION_TYPES})
-		if (CLANG_COMPILER_HEADER_SEARCH_PATH STREQUAL "")
-			set (CLANG_COMPILER_HEADER_SEARCH_PATH "${LLVM_BINARY_DIR}/${LLVM_CONFIGURATION_TYPE}/lib/clang/${LLVM_VERSION}/include")
-			message (STATUS "Trying to find Clang compiler headers in '${LLVM_CONFIGURATION_TYPE}' build config in directory '${CLANG_COMPILER_HEADER_SEARCH_PATH}'.")
+	set (CLANG_COMPILER_HEADER_SEARCH_PATH "/Users/me/Applications/llvm-project-master/binstall/lib/clang")
 			if (EXISTS ${CLANG_COMPILER_HEADER_SEARCH_PATH})
 				message (STATUS "Found headers for '${LLVM_CONFIGURATION_TYPE}' build config.")
 				file(GLOB_RECURSE CLANG_COMPILER_HEADER_PATHS RELATIVE "${CLANG_COMPILER_HEADER_SEARCH_PATH}" "${CLANG_COMPILER_HEADER_SEARCH_PATH}/*")
@@ -149,8 +145,6 @@ if (BUILD_CXX_LANGUAGE_PACKAGE)
 			else ()
 				set (CLANG_COMPILER_HEADER_SEARCH_PATH "")
 			endif ()
-		endif ()
-	endforeach()
 
 	if (CLANG_COMPILER_HEADER_SEARCH_PATH STREQUAL "")
 		message(FATAL_ERROR "Unable to copy Clang compiler headers from clang build dir.")
```

**Comment by mlangkabel** [comment]
This issue should be fixed by merge of pull request #1064.

**Comment by mlangkabel** [comment]
@anktkr: your issue seems to be the `LLVM_VERSION` variable, which somehow is set to `11.0.0git` by LLVM, instead of `11.0.0`. However, LLVM 11 has not yet been released, so this may be the cause for this issue.
